### PR TITLE
Attempt to fix flaky history revision test

### PR DESCRIPTION
**Pull request by uladzimirdev**
## Description

Attempt to fix flaky history revision cypress spec. 

image

In flaky runs at CI ([example](https://www.deploysentinel.com/ci/runs/64639aedb34905a540a5313b)) `/api/revision` returns the only record about dashboard creating, but not about adding a card to the dashboard. The reason for it *can be* an async nature of adding revision events on BE. 

In this PR we add a retry option for requests to the `/api/revision` until BE responds with 2 events

### At CI - 1 event in response
Screenshot 2023-05-23 at 13 38 44
Screenshot 2023-05-23 at 13 38 51

### Locally - 2 events in response
image (1)

### CODE REVIEW

**Review by WiNloSt — APPROVED** [review]
Thank you for fixing the flake 🙇

**Review by nemanjaglumac — CHANGES_REQUESTED** [review]
Let's not merge this until it passes a stress-test.
https://github.com/metabase/metabase/actions/workflows/e2e-stress-test-flake-fix.yml

File: e2e/support/helpers/e2e-dashboard-helpers.js
Comment (uladzimirdev) [reply e2e/support/helpers/e2e-dashboard-helpers.js:112]: I followed the similar concept from another file, I believe it makes sense to separate it as it checks body length, not that big abstraction, but thanks for the point. 

The nature of revision events is async in backend so I expect to see more problems like this in the future in other related tests

**Comment by uladzimirdev** [comment]
https://www.deploysentinel.com/ci/runs/646e282f61b9189af3a782fe 😱

**Comment by nemanjaglumac** [comment]
This is helping to a certain extent, but I'm not convinced that the test helper is the way to go. As you can see from the logs, even with this retry mechanism in place, this test will still fail quite often:
image

Something happened on FE or between FE and BE recently that made this test flake like crazy.
I think we should take a look into the root cause of it and fix it there, rather than trying to adjust the test to hide that root problem.

**Comment by uladzimirdev** [comment]
> This is helping to a certain extent, but I'm not convinced that the test helper is the way to go. As you can see from the logs, even with this retry mechanism in place, this test will still fail quite often: image
> 
> Something happened on FE or between FE and BE recently that made this test flake like crazy. I think we should take a look into the root cause of it and fix it there, rather than trying to adjust the test to hide that root problem.

true. even bigger timeout didn't help, so something is definitely went wrong and we need to find a root cause

**Comment by uladzimirdev** [comment]
As we need to continue investigation (whether it's FE-BE connection or just BE problem), I'm closing this

**Pull request closed without merging**
